src/github/review-pull-request.ts

```diff
@@ -0,0 +1,327 @@
+  'application/vnd.github.comfort-fade-preview+json';
+
+/**
+ * Multiline comment GitHub API parameters
+ * For more information see
+ * https://developer.github.com/v3/pulls/comments/#create-a-review-comment-for-a-pull-request
+ */
+interface MultilineComment {
+  body: string;
+  path: string;
+  start_line: number;
+  line: number;
+  side: 'RIGHT' | 'LEFT';
+  start_side: 'RIGHT' | 'LEFT';
+}
+
+interface SingleLineComment {
+  body: string;
+  path: string;
+  line: number;
+  side: 'RIGHT' | 'LEFT';
+}
+
+type Comment = SingleLineComment | MultilineComment;
+
+/**
+ * GitHub-defined type. The Octokit library/docs are probably behind since create review already
+ * accept multi-line code comments. However, the API does not reflect that.
+ */
+export type PullsCreateReviewParamsComments = {
+  path: string;
+  position: number;
+  body: string;
+};
+
+/**
+ * Convert the patch suggestions into GitHub parameter objects.
+ * Use this to generate review comments
+ * For information see:
+ * https://developer.github.com/v3/pulls/comments/#create-a-review-comment-for-a-pull-request
+ * @param suggestions
+ */
+export function buildReviewComments(
+  suggestions: Map<string, Hunk[]>
+): Comment[] {
+  const fileComments: Comment[] = [];
+  suggestions.forEach((hunks: Hunk[], fileName: string) => {
+    hunks.forEach(hunk => {
+      const newContent = hunk.newContent.join('\n');
+      if (hunk.oldStart === hunk.oldEnd) {
+        const singleComment: SingleLineComment = {
+          path: fileName,
+          body: `\`\`\`suggestion\n${newContent}\n\`\`\``,
+          line: hunk.oldEnd,
+          side: 'RIGHT',
+        };
+        fileComments.push(singleComment);
+      } else {
+        const comment: MultilineComment = {
+          path: fileName,
+          body: `\`\`\`suggestion\n${newContent}\n\`\`\``,
+          start_line: hunk.oldStart,
+          line: hunk.oldEnd,
+          side: 'RIGHT',
+          start_side: 'RIGHT',
+        };
+        fileComments.push(comment);
+      }
+    });
+  });
+  return fileComments;
+}
+
+/**
+ * Make a request to GitHub to make review comments
+ * @param octokit an authenticated octokit instance
+ * @param suggestions code suggestions patches
+ * @param remote the repository domain
+ * @param pullNumber the pull request number to make a review on
+ */
+export async function makeInlineSuggestions(
+  octokit: Octokit,
+  suggestions: Map<string, Hunk[]>,
+  outOfScopeSuggestions: Map<string, Hunk[]>,
+  remote: RepoDomain,
+  pullNumber: number
+): Promise<number | null> {
+  const comments = buildReviewComments(suggestions);
+  if (!comments.length) {
+    logger.info('No valid suggestions to make');
+  }
+  if (!comments.length && !outOfScopeSuggestions.size) {
+    logger.info('No suggestions were generated. Exiting...');
+    return null;
+  }
+  const summaryComment = buildSummaryComment(outOfScopeSuggestions);
+  if (summaryComment) {
+    logger.warn('Some suggestions could not be made');
```

Comment:
Perhaps this could be more descriptive

Reply:
This PR is only moving code around (and fixing the tests from proxyquire -> sinon). Filed #211 for this error message which we can do separately.